packages/decoder/test/current/test/downgrade-test.js

```diff
@@ -290,6 +290,36 @@ describe("Graceful degradation when information is missing", function () {
     assert.strictEqual(decodedFunction.value.selector, selector);
   });
 
+  it("Decodes transactions sent to the instance via additionalContexts", async function () {
+    let mangledCompilations = clonedeep(compilations);
+    let downgradeTest = mangledCompilations[0].contracts.find(
+      contract => contract.contractName === "DowngradeTest"
+    );
+    downgradeTest.deployedBytecode = undefined;
+
+    let deployedContract = await abstractions.DowngradeTest.new();
```

Comment:
Why does `deployedContract` not have a `deployedBytecode` field after deployment? And what is the point of setting `downgradeTest.deployedBytecode` to `undefined` because `downgradeTest`? It is not used in this test after that assignment.

Reply:
> Why does deployedContract not have a deployedBytecode field after deployment?

It does, but that's not where the decoder is going to look for it. :)  It's going to look in the project info, that it was deleted from.

> And what is the point of setting downgradeTest.deployedBytecode to undefined because downgradeTest? It is not used in this test after that assignment.

It is, just not explicitly!  It's part of `mangledCompilations`; `downgradeTest` is part of that, so the purpose of that line is to alter `mangledCompilations`.

Reply:
Oh ok I see. But when I print the value of `deployedContract.deployedBytecode` I get `undefined` in this test.

Reply:
...huh.  I guess there's some aliasing going on there that I forgot about...?